gl/gloffscreen_glx.c

```diff
@@ -95,17 +96,46 @@ GloContext *glo_context_create(int formatFlags)
 #endif
 
     /* Create GLX context */
-    context->glx_context = glXCreateNewContext(x_display, configs[0], GLX_RGBA_TYPE, NULL, True);
+    PFNGLXCREATECONTEXTATTRIBSARBPROC glXCreateContextAttribsARB =
+        (PFNGLXCREATECONTEXTATTRIBSARBPROC)
+            glXGetProcAddress((const GLubyte*)"glXCreateContextAttribsARB");
+    if (glXCreateContextAttribsARB == NULL) {
+        printf("GLX doesn't support ARB_create_context extension.\n");
+        exit(1);
+    }
+    int context_attribute_list[] = {
+        GLX_CONTEXT_MAJOR_VERSION_ARB, 3,
+        GLX_CONTEXT_MINOR_VERSION_ARB, 3,
+        GLX_CONTEXT_PROFILE_MASK_ARB, GLX_CONTEXT_CORE_PROFILE_BIT_ARB,
+        None
+    };
+    context->glx_context = glXCreateContextAttribsARB(x_display, configs[0], 0, True, context_attribute_list);
+    XSync(x_display, False);
     if (context->glx_context == NULL) return NULL;
     glo_set_current(context);
 
     if (!initialized) {
         /* Initialize glew */
+        glewExperimental = GL_TRUE;
         if (GLEW_OK != glewInit()) {
             /* GLEW failed! */
-            printf("Glew init failed.");
+            printf("GLEW init failed.\n");
+            exit(1);
+        }
+
+        /* Get rid of GLEW errors */
+        while(glGetError() != GL_NO_ERROR);
+
+        GLint major, minor;
+        glGetIntegerv(GL_MAJOR_VERSION, &major);
+        glGetIntegerv(GL_MINOR_VERSION, &minor);
+        printf("GL %d.%d\n",major,minor);
```

Comment:
Yes. glewInit will cause errors currently and they will assert() in the nv2a code otherwise.
I'll replace the version printf with an assert if you want.